### PR TITLE
[Eng] Verify Changelogs

### PR DESCRIPTION
**Pull request by seankane-msft**

### CODE REVIEW

File: eng/pipelines/templates/steps/analyze.yml [hunk 1]
```diff
@@ -46,3 +46,9 @@ steps:
       CheckLinkGuidance: $true
 
   - template: /eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml
+
+  - template: /eng/common/pipelines/templates/steps/verify-changelog.yml
```
Comment (chidozieononiwu) [inline eng/pipelines/templates/steps/analyze.yml:50]: It seems there is exactly one package per service for go. Is that always the case. If not you will have to run this for each artifact

**Review by chidozieononiwu — APPROVED** [review]

File: sdk/azcore/ci.yml [hunk 2]
```diff
@@ -15,5 +15,6 @@ pr:
 stages:
 - template: ../../eng/pipelines/templates/jobs/archetype-sdk-client.yml
   parameters:
+    PackageName: 'azcore'
```
Comment (weshaggard) [inline sdk/azcore/ci.yml:18]: Where did you use this parameter?

File: eng/pipelines/templates/steps/analyze.yml (hunk 1 above)
Comment (weshaggard) [reply eng/pipelines/templates/steps/analyze.yml:50]: In the case of azcore and azidentity I suspect you want either the serviceName or the packageName to be empty given there is only one directory layer.

File: eng/pipelines/templates/steps/analyze.yml [hunk 3]
```diff
@@ -54,3 +54,9 @@ steps:
...
+    parameters:
+      PackageName: ${{parameters.ServiceDirectory}}
+      ServiceName: ${{parameters.ServiceDirectory}}
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/analyze.yml:61]:
```suggestion
```

File: eng/pipelines/templates/jobs/archetype-go-release.yml [hunk 4]
```diff
@@ -24,10 +24,14 @@ stages:
                 steps:
                   - checkout: self
                   - template: /eng/common/pipelines/templates/steps/retain-run.yml
+                  - template: /eng/common/pipelines/templates/steps/verify-changelog.yml
+                    parameters:
+                      PackageName: ${{artifact.name}}
+                      ServiceName: ${{parameters.ServiceDirectory}}
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-go-release.yml:30]:
```suggestion
                      PackageName: ${{parameters.ServiceDirectory}}
```

File: eng/pipelines/templates/jobs/archetype-go-release.yml [hunk 5]
```diff
@@ -24,10 +24,13 @@ stages:
                 steps:
                   - checkout: self
                   - template: /eng/common/pipelines/templates/steps/retain-run.yml
+                  - template: /eng/common/pipelines/templates/steps/verify-changelog.yml
+                    parameters:
+                      PackageName: ${{parameters.ServiceDirectory}}
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-go-release.yml:29]:
```suggestion
                      PackageName: ${{parameters.ServiceDirectory}}
                      ServiceDirectory: ''
```

File: eng/pipelines/templates/steps/analyze.yml [hunk 6]
```diff
@@ -54,3 +54,8 @@ steps:
       CheckLinkGuidance: $true
 
   - template: /eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml
+
+  - template: /eng/common/pipelines/templates/steps/verify-changelog.yml
+    parameters:
+      PackageName: ${{parameters.ServiceDirectory}}
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/analyze.yml:60]:
```suggestion
      ServiceDirectory: ''
      PackageName: ${{parameters.ServiceDirectory}}
```

File: eng/pipelines/templates/steps/analyze.yml [hunk 7]
```diff
@@ -54,3 +54,9 @@ steps:
       CheckLinkGuidance: $true
 
   - template: /eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml
+
+  - template: /eng/common/pipelines/templates/steps/verify-changelog.yml
+    parameters:
+      ServiceDirectory: ''
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/analyze.yml:60]:
```suggestion
      ServiceName: ''
```

File: eng/pipelines/templates/jobs/archetype-go-release.yml [hunk 8]
```diff
@@ -24,10 +24,14 @@ stages:
                 steps:
                   - checkout: self
                   - template: /eng/common/pipelines/templates/steps/retain-run.yml
+                  - template: /eng/common/pipelines/templates/steps/verify-changelog.yml
+                    parameters:
+                      ServiceDirectory: ''
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-go-release.yml:29]:
```suggestion
                      ServiceName: ''
```
```suggestion
                      ServiceDirectory: ''
```

File: eng/pipelines/templates/jobs/archetype-go-release.yml [hunk 9]
```diff
@@ -24,10 +24,14 @@ stages:
                 steps:
                   - checkout: self
                   - template: /eng/common/pipelines/templates/steps/retain-run.yml
+                  - template: /eng/common/pipelines/templates/steps/verify-changelog.yml
+                    parameters:
+                      ServiceName: ''
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-go-release.yml:29]: @chidozieononiwu this is a workaround until https://github.com/Azure/azure-sdk-tools/pull/2016 can get merged.

Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-go-release.yml:29]:
```suggestion
```

File: eng/pipelines/templates/steps/analyze.yml [hunk 10]
```diff
@@ -54,3 +54,9 @@ steps:
       CheckLinkGuidance: $true
 
   - template: /eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml
+
+  - template: /eng/common/pipelines/templates/steps/verify-changelog.yml
+    parameters:
+      ServiceName: ''
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/analyze.yml:60]:
```suggestion
```

File: eng/pipelines/templates/jobs/archetype-go-release.yml (hunk 5 above)
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-go-release.yml:29]:
```suggestion
                      PackageName: 'sdk/${{parameters.ServiceDirectory}}'
```

File: eng/pipelines/templates/steps/analyze.yml (hunk 6 above)
Comment (weshaggard) [inline eng/pipelines/templates/steps/analyze.yml:60]:
```suggestion
      PackageName: 'sdk/${{parameters.ServiceDirectory}}`
```

File: eng/pipelines/templates/steps/analyze.yml [hunk 11]
```diff
@@ -54,3 +54,8 @@ steps:
       CheckLinkGuidance: $true
 
   - template: /eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml
+
+  - template: /eng/common/pipelines/templates/steps/verify-changelog.yml
+    parameters:
+      PackageName: 'sdk/${{parameters.ServiceDirectory}}`
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/analyze.yml:60]:
```suggestion
      PackageName: 'sdk/${{parameters.ServiceDirectory}}'
```

**Review by weshaggard — APPROVED** [review]

**Pull request merged**